apps/server/src/modules/user-login-migration/service/school-migration.service.ts

```diff
@@ -135,10 +133,14 @@ export class SchoolMigrationService {
 		await this.schoolService.save(schoolDO);
 	}
 
-	private async rollbackMigration(originalSchoolDO: SchoolDO) {
-		if (originalSchoolDO) {
-			await this.schoolService.save(originalSchoolDO);
-		}
+	async rollbackMigration(schoolId: EntityId, targetSystemId: string): Promise<void> {
```

Comment:
This rollback function is not quite right.
Its original purpose is to roll back the entire entity in case something goes wrong during the save operation, but this rollback pattern was meant as a replacement for transactions, which is actually not needed here. In the case of failure during saving to the database it would load the old school (the one that was not saved) now and rolling back on the old school, which would be wrong.
Furthermore, in our case now we want to cancel a migration that has not happened yet, so this would try to roll back on an unmigrated school, which would be wrong too.

Reply:
Yes, the rollback on the old school is a good point